### PR TITLE
[Clp] Add 1.17.3

### PR DESCRIPTION
**Pull request by ViralBShah**
Add support for mumps

### CODE REVIEW

**Comment by odow** [comment]
Maybe now now, but we can probably drop the ASL dependency. It looks optional (https://github.com/coin-or-tools/homebrew-coinor/blob/622c695c0f731dd2ae8535bb2612272ece43a3bb/clp.rb#L51), and I assume it's just for reading AMPL models which we don't do.

File: C/Coin-OR/Clp/build_tarballs.jl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 using BinaryBuilder, Pkg
 
 name = "Clp"
-version = v"1.17.5"
+version = v"1.17.3"
```
Comment (odow) [inline C/Coin-OR/Clp/build_tarballs.jl:4]: Do you have to update the commit as well? It's `27f61df3a85e327b8f7f15f8f40b0f415f4cc4e9`

Comment (ViralBShah) [reply C/Coin-OR/Clp/build_tarballs.jl:4]: Thanks! Nice catch.

**Comment by ViralBShah** [comment]
Can I drop ASL across the board? I'll do it when I am adding Mumps.

**Comment by odow** [comment]
Barring unindented consequences, the only one that needs ASL is Ipopt. Clp and Cbc don't.

**Comment by ViralBShah** [comment]
@odow I think I can also remove the Metis dependency. It seems looking at the Clp source that itis only used for Pardiso solvers. It would be nice to enable Pardiso at some point (since we do have the MKLPardiso.jl package), but for now, perhaps I can just drop it.

I suspect we still want Mumps. Can you confirm? It seemed that some benchmarks were slower without it.

**Comment by odow** [comment]
I wonder if we can also drop OSI. It's just for linking solvers together (It's COIN-OR's JuMP/MOI interface layer).

Yes, we probably still want Mumps. I'll re-run the benchmark

**Comment by ViralBShah** [comment]
Darn, I didn't know that. I thought it was essential. I'm all for dropping stuff. :-) 

I'll drop it in the next round.

**Comment by ViralBShah** [comment]
Actually I don't think we can drop it. It is required for building Cbc.

**Comment by odow** [comment]
Ah. Good point. Cbc calls out to Clp through it. Technically, it's possible to substitute other LP solvers into Cbc, but I'm not sure how easy that it.

**Comment by ViralBShah** [comment]
Do you know if we can link mumps 5 instead of mumps 4 which is what Coin-OR folks use. 

From https://coin-or.github.io/Ipopt/INSTALL.html as @dpo helpfully pointed out:

MUMPS will perform better with METIS, see METIS (matrix ordering).
MUMPS uses internally a fake implementation of MPI. If you are using Ipopt within an MPI program together with MUMPS, the code will not run. You will have to modify the MUMPS sources so that the MPI symbols inside the MUMPS code are renamed.
Branch mumps5 of project ThirdParty-Mumps can be used to build a a library of MUMPS 5.2.x that is usable with Ipopt. However, initial experiments on the CUTEst testset have shown that performance with MUMPS 5.2.x is worse than with Mumps 4.10.0 on average.

**Pull request merged**